template_app/src/charts/ApiBarChart.svelte

```diff
@@ -0,0 +1,43 @@
+<script lang="ts">
+  // Example from: https://www.chartjs.org/docs/latest/getting-started/usage.html
+  import { Chart } from "chart.js/auto";
+  import { getAquisitionsByYear } from "./api";
+  import { onMount } from "svelte";
+  export let title: string;
+  let data: any;
+  onMount(async () => {
+    data = await getAquisitionsByYear();
+    const options: any = {
+      type: "bar",
+      options: {
+        animation: false,
+        plugins: {
+          legend: {
+            display: false,
+          },
+          tooltip: {
+            enabled: false,
+          },
+        },
+      },
+      data: {
+        labels: data.map((row: any) => row.year),
+        datasets: [
+          {
+            label: "Acquisitions by year",
+            data: data.map((row: any) => row.count),
+          },
+        ],
+      },
+    };
+    new Chart(
+      (document.getElementById("myCanvas") as HTMLCanvasElement).getContext(
```

Comment:
This is correct / safe as it is, because `onMount` runs after the DOM elements below the script section are ready. But optionally, you could use https://learn.svelte.dev/tutorial/actions and/or a `let canvas: HTMLCanvasElement;` and `<canvas bind:this={canvas} />` instead of `getElementById`. The danger of `id` is that the IDs need to be unique for the whole page, so no other component can use this ID, or you can't have two `ApiBartChart` components on one page.

In fact, looks like `BarChart.svelte` is using actions as I'd expect. Could we do the same thing here to nudge people towards best practices? The complication is that the API results won't be immediately ready... Hmm, the simplest approach I know is to keep `onMount` as you have it, but get rid of `id` (for the uniqueness problem) and use `bind:this` instead

And it could be worth it to have a comment about why doing this instead of the `use:` actions is helpful, to guide people to doing the appropriate thing

Reply:
Thanks for the explanation and thoughts on this. I've updated as you suggested with a comment and the `bind:this` approach.